src/ansiblelint/app.py

```diff
@@ -324,13 +325,15 @@ def _sanitize_list_options(tag_list: list[str]) -> list[str]:
     return sorted(set(tags))
 
 
-@lru_cache(maxsize=1)
```

Comment:
Maybe we should bump this to `@lru_cache(maxsize=2)` which would cover both offline=True and offline=False. This would mean there are at most two app instances. It would also mean that syntax_check and _role_path can continue to request offline mode, and the expense of setting up the app for that only happens one extra time. Would that resolve the issue you're experiencing?

With this PR, only the first invocation of `get_app` will use the `offline` arg, otherwise the arg will be ignored.

Reply:
I am still playing with it. I think that I will remove the parameter as I am not aware of any real "production" use case where you would want to instantiate more than one and with different types. When you run it, you will either want online or offline mode.